lib/client.js

```diff
@@ -28,39 +35,84 @@ bedrock.events.on('bedrock.test.configure', function() {
   require('./test.config');
 });
 
+// TODO: Need validation on client objects
 bedrock.events.on('bedrock-mongodb.ready', function(callback) {
   if(!config['messages-client'].enable) {
     return callback();
   }
-  var interval = 1;
-  config['messages-client'].clients
-    .forEach(function(clientOptions) {
-      var jobId = createJobId(clientOptions.endpoint);
-      var endpoint = clientOptions.endpoint;
-      var privateKeyPem = clientOptions.privateKeyPem;
-      var publicKeyId = clientOptions.publicKeyId;
-      var strictSSL = !('strictSSL' in clientOptions) ?
-        true : !!clientOptions.strictSSL;
-      var fn = function(job, callback) {
-        var options = {
-          job: job,
-          endpoint: endpoint,
-          privateKeyPem: privateKeyPem,
-          publicKeyId: publicKeyId,
-          strictSSL: strictSSL
+  async.auto({
+    openCollections: function(callback) {
+      database.openCollections(['messageClient'], callback);
+    },
+    createIndexes: ['openCollections', function(callback) {
+      database.createIndexes([{
+        collection: 'messageClient',
+        fields: {id: 1},
+        options: {unique: true, background: false}
+      }], callback);
+    }],
+    insertClients: ['createIndexes', function(callback) {
+      var clients = config['messages-client'].clients;
+      async.each(clients, function(client, callback) {
+        var now = Date.now();
+        var meta = {};
+        meta.created = now;
+        meta.updated = now;
+        meta.recentPollErrorCount = 0;
+        var record = {
+          id: database.hash(client.id),
+          meta: meta,
+          client: client
         };
-        pollServer(options, function(err, results) {
-          if(err) {
-            return callback(err);
+        database.collections.messageClient.insert(
+        record, database.writeOptions, function(err, result) {
```

Comment:
Move the code for creating the record and inserting it into a function (doesn't have to be made public at this time). Also fix indentation, etc.